src/mod/pbkdf2.mod/pbkdf2.c

```diff
@@ -226,15 +226,39 @@ static char *pbkdf2_verify(const char *pass, const char *encrypted)
 static int tcl_encpass2 STDVAR
 {
   BADARGS(2, 2, " string");
-  if (strlen(argv[1]) > 0)
-    Tcl_AppendResult(irp, pbkdf2_encrypt(argv[1]), NULL);
-  else
-    Tcl_AppendResult(irp, "", NULL);
+  Tcl_SetResult(irp, pbkdf2_encrypt(argv[1]), TCL_STATIC);
+  return TCL_OK;
+}
+
+static int tcl_pbkdf2 STDVAR
+{
+  unsigned int rounds;
+  const EVP_MD *digest;
+  int digestlen;
+  unsigned char buf[256];
+
+  BADARGS(5, 5, " pass salt rounds digest");
+  rounds = atoi(argv[3]);
+  digest = EVP_get_digestbyname(argv[4]);
+  if (!digest) {
+    Tcl_AppendResult(irp, "PBKDF2 error: Unknown message digest '", argv[4], "'.", NULL);
+    return TCL_ERROR;
+  }
+  digestlen = EVP_MD_size(digest);
+  if (!PKCS5_PBKDF2_HMAC(argv[1], strlen(argv[1]), (const unsigned char *) argv[2], strlen(argv[2]), rounds, digest, digestlen, buf)) {
+    Tcl_AppendResult(irp, "PBKDF2 error: PKCS5_PBKDF2_HMAC(): ", ERR_error_string(ERR_get_error(), NULL), ".", NULL);
+    return TCL_ERROR;
+  }
+  buf[digestlen] = 0;
+  Tcl_Obj *result = Tcl_NewByteArrayObj(buf, digestlen);
```

Comment:
LOVE that you're using Objects! In my experience, I'm used to seeing hashes as a hex or base64 string instead of a byte array, and for encpass2 we're storing it as a base64 string... Can we do that here as well (base64 string) instead of returning this as a byte array, or does it need to be in byte format for a specific purpose? I'm trying to think through use cases and I think base64 will be more useful (and easier to handle/store) than a byte array

Reply:
there are difference conventions about padding regarding base64. also this function was about replacing the original tcl function, hence the same return type.

Reply:
meanwhile the code was changed to output hex str by default, bin optional.